File: src/main/java/com/bedirhanatasoy/game/tradingcard/rule/Rule.java
package com.bedirhanatasoy.game.tradingcard.rule;

import com.bedirhanatasoy.game.tradingcard.game.Game;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.function.Function;

@Getter
@AllArgsConstructor
class Rule {
    private RuleType type;
    private Function<Game, Boolean> condition;
    private Function<Game, Void> action;
    private Function<Game, Void> elseAction;
}

File: src/main/kotlin/com/bedirhanatasoy/game/card/rule/Rule.kt
package com.bedirhanatasoy.game.card.rule

import com.bedirhanatasoy.game.card.game.Game

class Rule(
    val type: RuleType,
    val condition: (Game) -> Boolean,
    val action: (Game) -> Unit,
    val elseAction: ((Game) -> Unit)? = null
)